gson/src/test/java/com/google/gson/functional/DefaultTypeAdaptersTest.java

```diff
@@ -101,6 +101,7 @@ public void testClassSerialization() {
   }
 
   @Test
+  @SuppressWarnings("GetClassOnClass")
   public void testClassDeserialization() {
     try {
       gson.fromJson("String.class", String.class.getClass());
```

Comment:
I thought the same thing, but when I removed the `.getClass()` the test failed.

Reply:
Just to make sure there is no misunderstanding, did you only remove the `.getClass()` (i.e. `fromJson(..., String.class)`) or replace the argument with `Class.class` (i.e. `fromJson(..., Class.class)`).

If I replace it with `Class.class` the test still seems to pass. Could you please share the assertion error message / exception stack trace (in case there is one)?

Reply:
I had removed just the `.getClass()`. Now, I have put the argument to `Class.class` and works fine.